src/common/turbo.ts

```diff
@@ -82,36 +129,41 @@ export type ArNSPurchaseReceipt = AoTokenCostParams & {
   createdDate: string;
 };
 
-export interface ArNSPaymentProvider {
+// Define separate provider interfaces
+export interface ArNSUnauthenticatedPaymentProvider {
```

Comment:
```suggestion
export interface ArNSPaymentProvider {
```

Reply:
So we would have ArNSPaymentProvider and ArNSAuthenticatedPaymentProvider?